libs/openchallenges/api-description/build/challenge.openapi.yaml

```diff
@@ -463,6 +463,22 @@ components:
       format: date
       nullable: true
       example: '2017-07-21'
+    EdamOperation:
```

Comment:
Nit: include `nullable: true`?

Reply:
I agree with your suggestion but this is a generated file that we can't edit manually.

Implementing the suggestion at the source level is a bit tricky. The reason is that `$ref:` doesn't combine well with `nullable:`, which is at least true for our version of OpenAPI (3.0.x).

There is a workaround with `AllOf` that I tried in the past, but this leads to issues with some generators (Python?). The alternative would be to add `nullable:` to the schema itself, i.e. `EdamOperation.yaml`. But a property may not always accept that its value may be `null`, so `nullable:` should ideally be defined independently for each variable, and not force all the variables to accept a `null` value, which would happen if we add `nullable: true` to `EdamOperation.yaml`. Does that make sense?

Recently I kind of settled on not specifying `nullable:`. Instead I used `required:` to specify whether a property should be expected. Here, `Challenge.operation` is not listed as required so the user should not expect to always receive this value, whether the property is missing or its value is `null`.

There are three actions that we could take:

1. Do nothing; keep the current implementation
2. Add `nullable: true` to `EdamOperation.yaml`: we only have one property that references this schema so why bother?
3. Create two versions of the schema: `EdamOperation.yaml` with `nullable: false` and `NullableEdamOperation.yaml` (or `EdamOperationOrNull.yaml`) with `nullable: true`.

Reply:
> Add nullable: true to EdamOperation.yaml: we only have one property that references this schema so why bother?

Adding `nullable: true` to the properties that can be null is the approach used with the other null-acceptable properties, e.g. SimpleChallengePlatform:

![Screenshot 2024-02-29 at 9 56 17 AM](https://github.com/Sage-Bionetworks/sage-monorepo/assets/9377970/4329e8f6-1ba6-404e-b8ad-766388eafd76)

My inclination would to stay consistent and go with option 2. (I would very much like to avoid option 3 😅 )

Reply:
@vpchung I added `nullable: true` to the OpenAPI schema `EdamOperation.yaml`